core/src/main/java/org/elasticsearch/action/support/replication/ReplicationOperation.java

```diff
@@ -50,6 +51,14 @@
             ReplicaRequest extends ReplicationRequest<ReplicaRequest>,
             PrimaryResultT extends ReplicationOperation.PrimaryResult<ReplicaRequest>
         > {
+
+    public static final Setting<ActiveShardCount> WAIT_FOR_ACTIVE_SHARDS_SETTING =
+        new Setting<>("action.wait_for_active_shards",
+                      (String) null,
+                      ActiveShardCount::parseString,
+                      Setting.Property.Dynamic,
+                      Setting.Property.NodeScope);
```

Comment:
+1 to index level. If possible, and unless it simplifies something here,  let's do it as a follow up - this PR is getting big :)